### PR TITLE
Chore: update docs on tagging a release

### PR DESCRIPTION
**Pull request by angela-tran**
We have been using annotated tags, so this updates our docs to reflect that.

### CODE REVIEW

File: docs/deployment/release.md [hunk 1]
```diff
@@ -87,7 +87,7 @@ git checkout prod
 
 git reset --hard origin/prod
 
-git tag YYYY.0M.R
+git tag -a YYYY.0M.R
```
Comment (lalver1) [inline docs/deployment/release.md:91]:
```suggestion
```

Git will open your default text editor and prompt you for the tag annotation. For the tag annotation, use the title of the `release`-tagged Issue that kicked off the release. Finally, after closing the text editor:

```bash
```

**Review by lalver1 — COMMENTED** [review]
Thanks for creating this PR @angela-tran ! I added a few more details about the annotation message as a suggested change. The suggested change preview looks a little weird but hopefully the raw text is clear.

File: docs/deployment/release.md (hunk 1 above)
Comment (angela-tran) [reply docs/deployment/release.md:91]: Good idea to add more details on this! I see what you mean with the raw text. I'll make that change

Comment (angela-tran) [reply docs/deployment/release.md:94]: Done in https://github.com/cal-itp/benefits/pull/2104/commits/64c404b89579b1d43e6a4e48a64c03b17ccce4c2

**Review by thekaveman — APPROVED** [review]

File: docs/deployment/release.md (hunk 1 above)
Comment (lalver1) [reply docs/deployment/release.md:94]: Looks good!

**Pull request merged**